### PR TITLE
Fix some issues with articles comparing

### PR DESCRIPTION
**Pull request by athongintel**
- add handlebar helper function ifCond to have if with binary condition
- fix compare not show deprecated, action type, showing redudant separator
- fix compare set modification for acticles that differ only in version, filename, id

### CODE REVIEW

**Comment by athongintel** [comment]
I changed the name back to original in package.json

File: template/main.js [hunk 1]
```diff
@@ -598,7 +614,7 @@ require([
                 compare: compareEntry,
                 versions: articleVersions[group][name]
             };
-
+
```
Comment (NicolasCARPi) [inline template/main.js]: Please remove extraneous whitespaces

File: package.json [hunk 2]
```diff
@@ -71,4 +71,4 @@
       "after"
     ]
   }
-}
+}
```
Comment (NicolasCARPi) [inline package.json]: please add newline at end of file

File: .gitignore [hunk 3]
```diff
@@ -4,3 +4,4 @@ node_modules/
 tmp/
 npm-debug.log
 .idea
+.DS_Store
```
Comment (NicolasCARPi) [inline .gitignore]: this is developer specific (like the .idea above but that's not your fault) and should be in your .gitignore global file

File: template/main.js [hunk 4]
```diff
@@ -641,8 +657,17 @@ require([
             $content.find('.versions li.version a').on('click', changeVersionCompareTo);
 
             // select navigation
-            $('#sidenav li[data-group=\'' + group + '\'][data-name=\'' + name + '\'][data-version=\'' + currentVersion + '\']').addClass('has-modifications');
-
+            // if there are changes except in version then set modification
+            var different = false;
+            $.each(fields.article, function(field){
+                different = different || !fields.compare[field] || (['id', 'version', 'filename'].indexOf(field) <0 && !deepCompare(fields.compare[field], fields.article[field]));
+	            // console.log(field, fields.compare[field], fields.article[field], different);
```
Comment (NicolasCARPi) [inline template/main.js]: remove dead code

File: template/main.js [hunk 5]
```diff
@@ -641,8 +657,17 @@ require([
...
+            });
+            // console.log(different);
```
Comment (NicolasCARPi) [inline template/main.js]: rm dead code

File: template/utils/handlebars_helper.js [hunk 6]
```diff
@@ -10,6 +10,35 @@ define([
      * Should be replaced with a full markdown lib.
      * @param string text
      */
+
+    Handlebars.registerHelper('ifCond', function (v1, operator, v2, options) {
```
Comment (NicolasCARPi) [inline template/utils/handlebars_helper.js]: please add a comment explaining why this is needed and how it works

**Pull request closed without merging**